### PR TITLE
[julia] Update keywords, built_in and literals for julia 1.X.

### PR DESCRIPTION
**Pull request by fredrikekre**
### Changes

This updates the lists of `keyword`, `built_in` and `literal` for Julia 1.X. Julia 1.X was released over 2 years ago so (IMO) it should be safe to remove the old tokens that are not applicable any more, but I don't know what your policy is for such things.

### Checklist
- [x] Tests: I did not find any tests for Julia syntax and adding this is beyond my capabilities (Edit: found this https://github.com/highlightjs/highlight.js/blob/master/test/detect/julia/default.txt so I will update that).
- [x] Updated the changelog at `CHANGES.md`
- [x] Added myself to `AUTHORS.txt`, under Contributors

### CODE REVIEW

**Comment by fredrikekre** [comment]
CC @mortenpi @tlienart, you might be interested in this.

**Comment by joshgoebel** [comment]
Thanks for this!  While we're at can we go ahead and move the words into one-word per line arrays to include the visibility and ease of future maintenance?  IE:

```
LITERALS = [
// one word per line
]
```

```
keywords : {
  literal: LITERALS.join(" "),
  // ...
```

You can see examples with javascript.js and ecmascript.js...

**Comment by joshgoebel** [comment]
>  it should be safe to remove the old tokens that are not applicable any more, but I don't know what your policy is for such things.

We'd usually leave them if there is a chance they're still in use in the world or old code snippets but since these lists are auto-generated I don't mind if we drop the older keywords since as you say it's been quite some time.

**Comment by fredrikekre** [comment]
Thanks, I updated the PR.

File: test/detect/julia/default.txt [hunk 1]
```diff
@@ -1,21 +1,5 @@
 ### Types
 
-# Old-style definitions
-
```
Comment (joshgoebel) [inline test/detect/julia/default.txt:4]: If we're going to cut all this we need to add in 5-10 lines of code that uses some of the new keywords as well.

File: src/languages/julia.js [hunk 2]
```diff
@@ -2,95 +2,328 @@
...
+    'PROGRAM_FILE',
+    'QuickSort',
+    'RoundDown',
+    'RoundFromZero',
+    'RoundNearest',
+    'RoundNearestTiesAway',
+    'RoundNearestTiesUp',
+    'RoundToZero',
+    'RoundUp',
+    'VERSION',
```
Comment (joshgoebel) [inline src/languages/julia.js:97]:
```suggestion
    'VERSION|0',
```

Comment (joshgoebel) [reply src/languages/julia.js:97]: Sees like this could be found in almost anything, to lowering it's relevance.

**Comment by joshgoebel** [comment]
Awesome.  Almost there! :-)

File: src/languages/julia.js (hunk 2 above)
Comment (fredrikekre) [reply src/languages/julia.js:97]: Alright, this is for autodetection? Isn't this needed for e.g. `for` which exists in almost any language?

**Comment by fredrikekre** [comment]
Added some new stuff to `test/detect`, and also noticed that some duplicates where included by my scripts, so fixed that too.

**Review by joshgoebel — APPROVED** [review]

**Comment by joshgoebel** [comment]
@fredrikekre Thanks so much for contributing!

**Pull request merged**